### PR TITLE
VB-2074 cache support types and supported prisons

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
VB-2074 cache support types and supported prisons so that the data is not fetched from DB every time.

## What does this pull request do?

Caches support types and supported prisons and evicts the cache at regular intervals
## What is the intent behind these changes?

Performance improvements.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/ScheduledCacheEvictTask.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  @Scheduled(cron = "\${cache.evict.support-types.cron:0 0 */12 * * ?}")
+  @CacheEvict(value = ["support-types"], allEntries = true)
+  fun evictSupportTypesCache() {
+    LOG.debug("Evicting support types cache.")
+  }
+
+  @Scheduled(cron = "\${cache.evict.supported-prisons.cron:0 0 */3 * * ?}")
+  @CacheEvict(value = ["supported-prisons"], allEntries = true)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/ScheduledCacheEvictTask.kt:25]: very nice, this must be a new annotation,..

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/config/GetSupportedPrisonsTest.kt [hunk 2]
```diff
@@ -40,9 +48,49 @@ class GetSupportedPrisonsTest : IntegrationTestBase() {
...
+    responseSpec = webTestClient.get().uri(SUPPORTED_PRISONS)
+      .headers(setAuthorisation(roles = requiredRole))
+      .exchange()
+
+    // Then
+    returnResult = responseSpec.expectStatus().isOk
+      .expectBody()
+    results = getResults(returnResult)
+    assertThat(results.size).isEqualTo(5)
+    verify(prisonRepository, times(1)).getSupportedPrisons()
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/config/GetSupportedPrisonsTest.kt:89]: nice

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/ScheduledCacheEvictTask.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/ScheduledCacheEvictTask.kt:25]: saw it here - https://www.baeldung.com/spring-boot-evict-cache

**Review by AledWEvans — COMMENTED** [review]
LGTM

**Pull request merged**